src/model/sprite.ts

```diff
@@ -0,0 +1,205 @@
+/*
+This file handles operations related to Sprites: loading them and getting info about them.
+A Sprite is a sub-image of a parent image.  The sub image is represented by a rectangle
+within the parent image (the four numbers: x, y, width, height).
+*/
+/*
+global
+G_view_drawSprite
+G_model_createCanvas
+*/
+
+type Sprite = [
+  HTMLCanvasElement | HTMLImageElement,
+  number,
+  number,
+  number,
+  number
+];
+
+const createSprite = (
+  img: HTMLCanvasElement | HTMLImageElement,
+  x: number,
+  y: number,
+  w: number,
+  h: number
+): Sprite => {
+  return [img, x, y, w, h];
+};
+
+// sprites are stored inside key => value objects, where the key is the name of the sprite
+type SpriteCollection = { [key: string]: Sprite };
+
+// This type represents the postfix one can add to a sprite
+type SpriteModification = '' | '_r1' | '_r2' | '_r3' | '_f';
+const G_SPRITE_MOD_NORMAL: SpriteModification = '';
+const G_SPRITE_MOD_FLIPPED: SpriteModification = '_f';
+const G_SPRITE_MOD_ROT90: SpriteModification = '_r1';
+const G_SPRITE_MOD_ROT180: SpriteModification = '_r2';
+const G_SPRITE_MOD_ROT270: SpriteModification = '_r3';
+
+let model_sprites: SpriteCollection | null = null;
+
+// given an inputCanvas, return a new canvas rotated to the right by 90 degrees
+const createRotatedImg = (
```

Comment:
This is a little confusing, but I get the gist.

Reply:
Images are drawn at a position `x, y`  This is the top left of the image.  When rotating a sprite, the focal point is the `x, y` position, which, if it's at the top left, means you rotate around the top left.  That's not what I want.  I want it to rotate around the center.  So first translate the focal point to the center of the image, then rotate, but we have to translate back or else the image will be drawn where the top left is at the focal point in the middle.